crates/uv/tests/it/sync.rs

```diff
@@ -3482,6 +3482,133 @@ fn no_install_project_no_build() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn virtual_no_build() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["anyio==3.7.0"]
+        "#,
+    )?;
+
+    // Generate a lockfile.
+    context.lock().assert().success();
+
+    // Clear the cache.
+    fs_err::remove_dir_all(&context.cache_dir)?;
+
+    // `--no-build` should not raise an error, since we don't install virtual projects.
+    uv_snapshot!(context.filters(), context.sync().arg("--no-build"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 4 packages in [TIME]
+    Prepared 3 packages in [TIME]
+    Installed 3 packages in [TIME]
+     + anyio==3.7.0
+     + idna==3.6
+     + sniffio==1.3.1
+    ");
+
+    Ok(())
+}
+
+#[test]
+fn virtual_no_build_dynamic_cached() -> Result<()> {
```

Comment:
Super confusing that this test succeeds but the next test fails. The difference is that in this test, the dynamic metadata is cached, so we don't need to run any Python.